### PR TITLE
Modernize, add Python 3 support

### PR DESCRIPTION
**Pull request by lwgordonimos**

### CODE REVIEW

File: aodncore/util/fileops.py [hunk 1]
```diff
@@ -137,7 +138,7 @@ def is_dir_writable(path):
     """
     try:
         with TemporaryFile(prefix='is_dir_writable', suffix='.tmp', dir=path) as t:
-            t.write('is_dir_writable')
+            t.write('is_dir_writable'.encode('utf-8'))
```
Comment (mhidas) [inline aodncore/util/fileops.py]: Is `'x'.encode('utf-8')` the same as `u'x'`?

**Comment by mhidas** [comment]
Is this ready to merge?

File: aodncore/pipeline/handlerbase.py [hunk 2]
```diff
@@ -527,13 +528,14 @@ def logger(self):
         return self._logger
 
     @lazyproperty
-    def module_versions(self):
+    def versions(self):
```
Comment (mhidas) [inline aodncore/pipeline/handlerbase.py]: This isn't called explicitly anywhere?

File: aodncore/vocab/platform_code_vocab.py [hunk 3]
```diff
@@ -80,7 +80,7 @@ def platform_type_uris_by_category(self):
                     # handle more than 1 url match per category of platform
                     if platform_element_sublabels is not None:
                         if 'narrowMatch' in platform_element_sublabels:
-                            val_cat_url = val.attrib.values()[0]
+                            val_cat_url = list(val.attrib.values())[0]
```
[the remark this answers is not in the document]
Comment (lwgordonimos) [reply aodncore/vocab/platform_code_vocab.py]: Yeah any number of these could be written differently, but for example, that would result in changing the exception type from IndexError to StopIteration, so the minimal "compatible" change was my priority here, without fixing or optimising anything at the same time.

File: aodncore/util/fileops.py (hunk 1 above)
Comment (lwgordonimos) [reply aodncore/util/fileops.py]: It actually needs a "bytes-like" object not a unicode. Could have been written as a b-string, but results in the same thing.

```python
In[20]: b'is_dir_writable' == 'is_dir_writable'.encode('utf-8')
Out[20]: True
```

File: aodncore/pipeline/handlerbase.py (hunk 2 above)
Comment (lwgordonimos) [reply aodncore/pipeline/handlerbase.py]: Nope, is only used when representing the handler as a string, i.e. in the logs

**Comment by lwgordonimos** [comment]
Yeah is mergable as far as I have tested, haven't found any issues so far spinning up on Python 2 or 3 in a test environment.

File: aodncore/util/fileops.py (hunk 1 above)
Comment (mhidas) [reply aodncore/util/fileops.py]: Ah! Right, thanks. Still getting my head around bytes/string/unicode objects.

**Pull request merged**